test/e2e/app/app.go

```diff
@@ -170,8 +171,8 @@ func (app *Application) updateVoteExtensionEnableHeight(currentHeight int64) *cm
 			"current_height", currentHeight,
 			"enable_height", app.cfg.VoteExtensionsEnableHeight)
 		params = &cmtproto.ConsensusParams{
-			Abci: &cmtproto.ABCIParams{
-				VoteExtensionsEnableHeight: app.cfg.VoteExtensionsEnableHeight,
+			Feature: &cmtproto.FeatureParams{
+				VoteExtensionsEnableHeight: &gogo.Int64Value{Value: app.cfg.VoteExtensionsEnableHeight},
```

Comment:
Can we use 0 to mean "no height specified" like in all other places where it is so?
This nullable JS integer field is clunky, it will be tedious to work with in Rust and probably other languages.

Reply:
We had infinite discussions on that respect.

The problem is that `0` is value with meaning for these parameters. It means disabled.

The fact that `0` is also the nil value for an integer, if the user does not set one of the values here, it would implicitly set it to `0`. And the validation rules will fail, because once enabled (from height `H > 0`), these features cannot be disabled.

Reply:
Let me throw in some less (or equally) cumbersome alternatives just in case they were not considered:

* Enable features with bit flags set in a separate field, so that a height field is only considered if the corresponding feature flag is set.

* Introduce a `ConsensusParamUpdate` message which has a `oneof` for all parameters that can be updated. This is to be used as a repeated field in places where `ConsensusParams` is currently used as a delta value with nils meaning no update. `ConsensusParams` then cleanly describes the current state.